lib/galaxy/managers/history_contents.py

```diff
@@ -397,9 +405,10 @@ def _subcontainer_id_map(self, id_list):
         component_class = self.subcontainer_class
         query = (self._session().query(component_class)
             .filter(component_class.id.in_(id_list))
-            .options(eagerload('collection'))
-            .options(eagerload('tags'))
-            .options(eagerload('annotations')))
+            .options(joinedload('collection'))
+            .options(joinedload('tags'))
+            .options(joinedload('annotations'))
+            .options(joinedload('job_state_summary')))
```

Comment:
Will this always happen now? Are we doing a huge join across that table for things we will only need if 'job_state_summary' is requested from the user - which the current client won't do? Or is this deferred in some way?

Reply:
It will happen on the secondary query which retrieves the collections which match the filtered content query.

There are 3 queries in the contents result, the union I described at the top of the PR, then two additional ones which return the details for HDA and HDCAs which match the filtering criteria which has already run

Thus the last 2 queries are a "select fields from hdca where id in 1,3,4,5" scenario where the IDS are collected from the first unioned query.

Reply:
Currently the existing classes do not pass in the user-requested serialization parameters into the content query generation class. Theoretically, could add that as a new parameter, which would allow the content query to conditionally decide whether or not to eager join stuff like this, but the new history client would ask for it every time.

If your goal is to make the new serialization option available to the new history client without unduly impacting the existing client's query, I could add a few extra noodles to that spaghetti code.

Reply:
@jmchilton I've added the additional parameter to the content query assembly to make sure that the eager load against the view only occurs if the user explicitly selects it.

The weakness of the serialization class scheme is evident here as the logic to evaluate whether or not to do the eager load is brittle now. If somebody decides to move the job_states_summary field into a pre-defined view then the logic will fail.

I think refactoring the serializers is in order but it can be done later.